src/main/java/org/opentripplanner/routing/impl/StreetVertexIndex.java

```diff
@@ -273,11 +273,11 @@ private TraverseMode getTraverseModeForLinker(RoutingRequest options, boolean en
     //It can be null in tests
     if (options != null) {
       TraverseModeSet modes = options.streetSubRequestModes;
-      if (modes.getCar()) {
-        // for park and ride we will start in car mode and walk to the end vertex
-        if (!endVertex && options.parkAndRide) {
-          nonTransitMode = TraverseMode.CAR;
-        }
+      // for park and ride we will start in car mode and walk to the end vertex
+      boolean parkAndRideDepart = modes.getCar() && options.parkAndRide && !endVertex;
+      boolean onlyCarAvailable = modes.getCar() && !(modes.getWalk() || modes.getBicycle());
+      if (onlyCarAvailable || parkAndRideDepart) {
```

Comment:
If I understand this PR correctly, this is the actual part that changes the linking behaviour, correct?

Reply:
Yes -- everything else is needed so that this is called with the correct parameters.